apps/omg_db/lib/omg_db/leveldb/core.ex

```diff
@@ -41,9 +41,10 @@ defmodule OMG.DB.LevelDBCore do
   @doc """
   Interprets the response from leveldb and returns a success-decorated result
   """
+  @spec decode_value({:ok, binary()} | :not_found, atom()) :: {:ok, term()} | :not_found
   def decode_value(db_response, type) do
     case decode_response(type, db_response) do
-      {:error, error} -> {:error, error}
+      :not_found -> :not_found
```

Comment:
I'm not exactly sure what you mean, but that particular case was dead code when I looked at the behavior.

Reply:
There are three calls into `decode_value/2` all coming from `LevelDB` module. The `ExLevelDB` contract is specified as 
```
@spec get(db_reference, db_key, read_options) :: {:ok, Bitstring} | :not_found
```

Reply:
So reading dialyzer, this would never have been reached: https://github.com/omisego/elixir-omg/pull/629/files#diff-94513f48bd6e51e3701e754a95a47384L103
Neither was it never reached in all tests.

Reply:
ok, digging a little deeper into the new code, I think the issue is really with the `decode_value` - for a key not found it should return `{:ok, :not_found}` and it did so, but now it would return just `:not_found`. `decode_values` seem to not have changed behavior, so :ok_hand: .

So it seems having `decode_value` just return whatever the lower-level `decode_response` returns will bring the behavior back.

Not sure why the dead code emerged - maybe `ExLevelDb` used to be able to return sth else or maybe some kind of overzealous handling of possible returns. If it's good enough to be prepared for what the `spec` tells us then ok.

Reply:
You want to return `{:ok, not_found}` instead of `:not_found`?

Reply:
I want the old behavior